runtime/v1/shim/service.go

```diff
@@ -503,47 +503,45 @@ func (s *Service) processExits() {
 	}
 }
 
-func (s *Service) allProcesses() []process.Process {
+func (s *Service) getProcess(pid int) process.Process {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	res := make([]process.Process, 0, len(s.processes))
 	for _, p := range s.processes {
-		res = append(res, p)
+		if p.Pid() == pid {
+			return p
+		}
 	}
-	return res
+	return nil
 }
 
 func (s *Service) checkProcesses(e runc.Exit) {
-	for _, p := range s.allProcesses() {
-		if p.Pid() != e.Pid {
-			continue
+	p := s.getProcess(e.Pid)
+	if p == nil {
```

Comment:
Maybe it make sense to log here something like "process with id:%d wasn't found"